mempool/src/pool/mod.rs

```diff
@@ -263,77 +269,78 @@ where
         // We don't have a notion of MAX_MONEY like Bitcoin Core does, so we also don't have a
         // `MoneyRange` check like the one found in Bitcoin Core's `CheckTransaction`
 
-        if tx.transaction().inputs().is_empty() {
-            return Err(TxValidationError::NoInputs);
-        }
+        // TODO: The following three checks may each better be discarded or moved to tx verifier
+        ensure!(
+            !tx.transaction().inputs().is_empty(),
+            TxValidationError::NoInputs,
+        );
 
-        if tx.transaction().outputs().is_empty() {
-            return Err(TxValidationError::NoOutputs);
-        }
+        ensure!(
+            !tx.transaction().outputs().is_empty(),
+            TxValidationError::NoOutputs,
```

Comment:
Isn't it ok to burn utxo this way?
If not I'll add such an error to `input_output_poliecy.rs` and we can get rid of this check here

Reply:
> Isn't it ok to burn utxo this way? If not I'll add such an error to `input_output_poliecy.rs` and we can get rid of this check here

Yea I am not sure. I am inclined to allow this but I don't claim I can work out all the security implications. Just as an aside, this does not burn the coins but rather pays them to the miner as a fee (even tokens when paying any token as a fee is implemented).

There's also distinction between consensus rules and mempool policy. The node may choose to not accept a transaction in the mempool but consider it valid if it appears in a block.

Reply:
I think this should be a decision coming from the tx-verifier.

We can have mempool rules, but then a question is whether want these rules done in a separate, clear function that clarifies this difference.

Reply:
> > We can have mempool rules, but then a question is whether want these rules done in a separate, clear function that clarifies this difference.
> 
> Having a separate function for this is on my todo list

Added some basic separation